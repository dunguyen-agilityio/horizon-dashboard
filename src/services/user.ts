// Models
import { TUser } from '@/models/User';

// Services
import { apiClient } from './api';

// Constants
import { API_ENTITY } from '@/constants';

export const getTeamMember = () =>
  apiClient.get<TUser[]>(
    `${API_ENTITY.USERS}?limit=3&populate[0]=avatar&populate[1]=role`,
  );
