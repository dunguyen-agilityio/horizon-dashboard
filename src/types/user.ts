// Model
import { TImage } from '@/models/Image';
import { TUser } from '@/models/User';

export type Role = { name: string };

export type UserExtend = Omit<TUser, 'avatar' | 'fullName'> & {
  avatar?: TImage;
};
