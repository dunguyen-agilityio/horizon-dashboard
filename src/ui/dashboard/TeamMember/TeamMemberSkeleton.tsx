import { Skeleton } from '@nextui-org/skeleton';

const TeamMemberSkeleton = () => (
  <Skeleton className="flex-1 h-full rounded-md bg-white dark:bg-indigo-light" />
);

export default TeamMemberSkeleton;
