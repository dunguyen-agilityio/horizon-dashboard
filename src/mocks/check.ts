import { TCheck } from '@/types/check';

export const MOCK_CHECKS: TCheck[] = [
  {
    id: '1',
    name: 'Tony Reichert',
    progress: 45,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '2',
    name: 'Zoey Lang',
    progress: 15,
    quantity: 20,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '3',
    name: 'Jane Fisher',
    progress: 55,
    quantity: 28,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
  {
    id: '4',
    name: 'William Howard',
    progress: 65,
    quantity: 30,
    createdAt: new Date('2024-08-14T08:25:04.188Z'),
  },
];
