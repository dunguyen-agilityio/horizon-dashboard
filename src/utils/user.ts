import { TUser, User } from '@/models/User';
import { UserResponse } from '@/types/contributor';
import { StrapiResponse } from '@/types/strapi';
import { formatStrapiModel } from './strapi';

export const formatUser = (data: TUser) => {
  const user = new User(data);

  return { ...user, fullName: user.username };
};

export const formatUserResponse = (
  data: StrapiResponse<UserResponse>,
): User => {
  const { avatar, role, ...rest } = formatStrapiModel(data);

  const { attributes: avatarData, id } = avatar?.data ?? {};

  const hasAvatar = avatarData && id;

  return formatUser({
    ...rest,
    role: { name: role?.attributes.name ?? '' },
    ...(hasAvatar && {
      avatar: { hash: avatarData.hash, url: avatarData.url },
    }),
  });
};
