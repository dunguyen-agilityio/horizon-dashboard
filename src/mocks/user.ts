import { TImage } from '@/models/Image';
import { TUser } from '@/models/User';
import { StrapiResponse } from '@/types/strapi';
import { UserResponse1 } from '@/utils/nft';

export const MOCK_AVATAR: TImage = {
  hash: 'image_de1e29b0b7',
  url: 'https://res.cloudinary.com/dn5vw6fjp/image/upload/v1726630608/image_de1e29b0b7.png',
};

export const MOCK_USERS: TUser[] = [
  {
    id: '1',
    username: 'Adela Parkson',
    firstName: 'User 1',
    email: 'user1@gmail.com',
    role: { name: 'Creative Director' },
    createdAt: new Date('2024-08-17T07:25:04.188Z'),
    rating: 29,
    avatar: MOCK_AVATAR,
  },
  {
    id: '2',
    username: 'user2',
    firstName: 'User 2',
    email: 'user2@gmail.com',
    role: { name: 'Product Designer' },
    createdAt: new Date('2024-08-13T07:25:04.188Z'),
    rating: 72,
    avatar: MOCK_AVATAR,
  },
  {
    id: '3',
    username: 'user3',
    firstName: 'User 3',
    email: 'user3@gmail.com',
    role: { name: 'Product Designer' },
    createdAt: new Date('2024-01-13T07:25:04.188Z'),
    rating: 89,
    avatar: MOCK_AVATAR,
  },
  {
    id: '4',
    username: 'user4',
    firstName: 'User 4',
    email: 'user4@gmail.com',
    role: { name: 'Junior Graphic Designer' },
    createdAt: new Date('2024-08-18T07:25:04.188Z'),
    rating: 52,
    avatar: MOCK_AVATAR,
  },
];

export const MOCK_ROLE_RESPONSE: StrapiResponse<{ id: string; name: string }> =
  {
    id: '1',
    attributes: { name: 'Developer' },
  };

export const MOCK_IMAGE_RESPONSE: StrapiResponse<TImage> = {
  id: '1',
  attributes: {
    hash: 'image_de1e29b0b7',
    url: 'https://res.cloudinary.com/dn5vw6fjp/image/upload/v1726630608/image_de1e29b0b7.png',
  },
};

export const MOCK_USERS_RESPONSE: StrapiResponse<UserResponse1>[] =
  MOCK_USERS.map(({ id, ...attributes }) => ({
    attributes: {
      ...attributes,
      avatar: { data: MOCK_IMAGE_RESPONSE },
      role: MOCK_ROLE_RESPONSE,
    },
    id,
  }));
