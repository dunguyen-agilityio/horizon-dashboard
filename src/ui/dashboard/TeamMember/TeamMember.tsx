'use client';

// Components
import { User } from '@nextui-org/user';
import { Listbox, ListboxItem } from '@nextui-org/listbox';
import { BoxIcon, Text } from '@/components';

// Models
import { UserResponse } from '@/models/User';

// Icons
import { Add, More } from '@/icons';

// Utils
import { formatUser } from '@/utils/user';

interface TeamMemberProps {
  members: UserResponse[];
}

const TeamMember = ({ members }: TeamMemberProps) => (
  <div className="w-full h-full bg-white dark:bg-indigo rounded-md pt-5 px-6 pb-[26px] sm:w-card md:w-auto flex flex-col gap-5">
    <div className="flex justify-between items-center h-8">
      <Text as="h2" className="text-lg leading-xl font-bold">
        Team Members
      </Text>
      <button
        className="p-[10px] rounded-[10px] bg-gray dark:bg-indigo-light disabled:cursor-not-allowed disabled:opacity-80"
        disabled
      >
        <div className="bg-primary dark:bg-white rounded-full w-6 h-6 flex items-center justify-center">
          <Add color="secondary" />
        </div>
      </button>
    </div>
    <Listbox
      items={members}
      classNames={{ list: 'gap-4', base: 'flex-1 px-4' }}
    >
      {(item) => {
        const { id, fullName, avatar, role } = formatUser(item);

        return (
          <ListboxItem
            key={id}
            classNames={{
              title: 'flex justify-between items-center px-4 py-3',
              base: 'py-0 ',
            }}
            className="w-full bg-white dark:bg-indigo-light shadow-light-card dark:shadow-none"
          >
            <User
              name={fullName}
              description={role}
              classNames={{
                name: 'text-medium font-bold',
                description: 'text-tiny font-medium text-secondary',
              }}
              avatarProps={{
                src: avatar,
                classNames: { base: 'w-[46px] h-[46px]' },
              }}
            />
            <button
              disabled
              className="disabled:cursor-not-allowed disabled:opacity-80"
            >
              <BoxIcon icon={<More />} customClass="fill-secondary" />
            </button>
          </ListboxItem>
        );
      }}
    </Listbox>
  </div>
);

export default TeamMember;
