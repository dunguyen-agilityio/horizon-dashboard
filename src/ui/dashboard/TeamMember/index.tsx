import { Text } from '@/components';
import { User } from '@nextui-org/user';
import { Listbox, ListboxItem } from '@nextui-org/listbox';
import { User as UserModel } from '@/models/User';
import { Add } from '@/icons';
import More from '@/icons/More';

interface TeamMemberProps {
  members: UserModel[];
}

const TeamMember = ({ members }: TeamMemberProps) => (
  <div className="bg-white dark:bg-indigo rounded-md p-7 w-full sm:w-[382px]">
    <div className="flex justify-between items-center">
      <Text as="h2" className="text-lg leading-xl font-bold">
        Team Members
      </Text>
      <button
        className="p-[10px] rounded-[10px] bg-gray dark:bg-indigo-light disabled:cursor-not-allowed disabled:opacity-80"
        disabled
      >
        <div className="bg-primary dark:bg-white rounded-full w-6 h-6 flex items-center justify-center">
          <Add color="secondary" />
        </div>
      </button>
    </div>
    <Listbox items={members} classNames={{ list: 'gap-4', base: 'pt-5' }}>
      {(item) => {
        const { avatar, id, fullName, role } = new UserModel(item);

        return (
          <ListboxItem
            key={id}
            classNames={{
              title: 'flex justify-between items-center px-4 py-3',
            }}
            className="w-full bg-white dark:bg-indigo-light shadow-light-card dark:shadow-none"
          >
            <User
              name={fullName}
              description={role}
              classNames={{
                name: 'text-medium font-bold',
                description: 'text-tiny font-medium text-secondary',
              }}
              avatarProps={{
                src: avatar,
                classNames: { base: 'w-[46px] h-[46px]' },
              }}
            />
            <button
              disabled
              className="disabled:cursor-not-allowed disabled:opacity-80"
            >
              <More />
            </button>
          </ListboxItem>
        );
      }}
    </Listbox>
  </div>
);

export default TeamMember;
