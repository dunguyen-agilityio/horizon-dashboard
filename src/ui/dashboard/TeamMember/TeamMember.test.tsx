import { render } from '@testing-library/react';
import TeamMember from '.';
import { MOCK_USERS } from '@/mocks/user';

describe('TeamMembers tests', () => {
  it('Should match snapshot', () => {
    const { container } = render(<TeamMember members={MOCK_USERS} />);
    expect(container).toMatchSnapshot();
  });
});
