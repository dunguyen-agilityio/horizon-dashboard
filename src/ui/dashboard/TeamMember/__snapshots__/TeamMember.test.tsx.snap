// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMembers tests Should match snapshot 1`] = `
<div>
  <div
    class="w-full h-full bg-white dark:bg-indigo rounded-md pt-5 px-6 pb-[26px] sm:w-card md:w-auto flex flex-col gap-5"
  >
    <div
      class="flex justify-between items-center h-8"
    >
      <h2
        class="text-primary dark:text-white text-lg leading-xl font-bold"
      >
        Team Members
      </h2>
      <button
        class="p-[10px] rounded-[10px] bg-gray dark:bg-indigo-light disabled:cursor-not-allowed disabled:opacity-80"
        disabled=""
      >
        <div
          class="bg-primary dark:bg-white rounded-full w-6 h-6 flex items-center justify-center"
        >
          <svg
            class="[&>path]:stroke-white [&>path]:dark:stroke-primary"
            fill="none"
            height="17"
            viewBox="0 0 19 17"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.32703 3.71619V13.2787"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
            <path
              d="M14.3896 8.49744H4.26459"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
            />
          </svg>
        </div>
      </button>
    </div>
    <div
      class="w-full relative flex flex-col gap-1 p-1 flex-1 px-4"
      data-slot="base"
    >
      <ul
        class="w-full flex flex-col outline-none gap-4"
        data-slot="list"
        id="react-aria-:r0:"
        role="listbox"
        tabindex="0"
      >
        <li
          aria-labelledby="react-aria-:r4:"
          class="flex group gap-2 items-center justify-between relative px-2 h-full box-border rounded-small subpixel-antialiased cursor-pointer tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[focus-visible=true]:dark:ring-offset-background-content1 data-[hover=true]:transition-colors data-[hover=true]:bg-default data-[hover=true]:text-default-foreground data-[selectable=true]:focus:bg-default data-[selectable=true]:focus:text-default-foreground py-0 w-full bg-white dark:bg-indigo-light shadow-light-card dark:shadow-none"
          data-key="1"
          id="react-aria-:r0:-option-1"
          role="option"
          tabindex="-1"
        >
          <span
            class="flex-1 text-small font-normal truncate flex justify-between items-center px-4 py-3"
            data-label="true"
            id="react-aria-:r4:"
          >
            <div
              class="inline-flex items-center justify-center gap-2 rounded-small outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              tabindex="-1"
            >
              <span
                class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-tiny bg-default text-default-foreground rounded-full w-[46px] h-[46px]"
                tabindex="-1"
              >
                <img
                  alt="avatar"
                  class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
                  src="/default-avatar.webp"
                />
              </span>
              <div
                class="inline-flex flex-col items-start"
              >
                <span
                  class="text-inherit text-medium font-bold"
                />
                <span
                  class="text-tiny font-medium text-secondary"
                >
                  Creative Director
                </span>
              </div>
            </div>
            <button
              class="disabled:cursor-not-allowed disabled:opacity-80"
              disabled=""
            >
              <i
                class="w-6 h-6 flex justify-center items-center hover:cursor-pointer fill-secondary"
              >
                <svg
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_201_3969)"
                  >
                    <path
                      d="M12 8C13.1 8 14 7.1 14 6C14 4.9 13.1 4 12 4C10.9 4 10 4.9 10 6C10 7.1 10.9 8 12 8ZM12 10C10.9 10 10 10.9 10 12C10 13.1 10.9 14 12 14C13.1 14 14 13.1 14 12C14 10.9 13.1 10 12 10ZM12 16C10.9 16 10 16.9 10 18C10 19.1 10.9 20 12 20C13.1 20 14 19.1 14 18C14 16.9 13.1 16 12 16Z"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_201_3969"
                    >
                      <rect
                        fill="white"
                        height="24"
                        width="24"
                      />
                    </clippath>
                  </defs>
                </svg>
              </i>
            </button>
          </span>
        </li>
        <li
          aria-labelledby="react-aria-:r6:"
          class="flex group gap-2 items-center justify-between relative px-2 h-full box-border rounded-small subpixel-antialiased cursor-pointer tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[focus-visible=true]:dark:ring-offset-background-content1 data-[hover=true]:transition-colors data-[hover=true]:bg-default data-[hover=true]:text-default-foreground data-[selectable=true]:focus:bg-default data-[selectable=true]:focus:text-default-foreground py-0 w-full bg-white dark:bg-indigo-light shadow-light-card dark:shadow-none"
          data-key="2"
          id="react-aria-:r0:-option-2"
          role="option"
          tabindex="-1"
        >
          <span
            class="flex-1 text-small font-normal truncate flex justify-between items-center px-4 py-3"
            data-label="true"
            id="react-aria-:r6:"
          >
            <div
              class="inline-flex items-center justify-center gap-2 rounded-small outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              tabindex="-1"
            >
              <span
                class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-tiny bg-default text-default-foreground rounded-full w-[46px] h-[46px]"
                tabindex="-1"
              >
                <img
                  alt="avatar"
                  class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
                  src="/default-avatar.webp"
                />
              </span>
              <div
                class="inline-flex flex-col items-start"
              >
                <span
                  class="text-inherit text-medium font-bold"
                />
                <span
                  class="text-tiny font-medium text-secondary"
                >
                  Product Designer
                </span>
              </div>
            </div>
            <button
              class="disabled:cursor-not-allowed disabled:opacity-80"
              disabled=""
            >
              <i
                class="w-6 h-6 flex justify-center items-center hover:cursor-pointer fill-secondary"
              >
                <svg
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_201_3969)"
                  >
                    <path
                      d="M12 8C13.1 8 14 7.1 14 6C14 4.9 13.1 4 12 4C10.9 4 10 4.9 10 6C10 7.1 10.9 8 12 8ZM12 10C10.9 10 10 10.9 10 12C10 13.1 10.9 14 12 14C13.1 14 14 13.1 14 12C14 10.9 13.1 10 12 10ZM12 16C10.9 16 10 16.9 10 18C10 19.1 10.9 20 12 20C13.1 20 14 19.1 14 18C14 16.9 13.1 16 12 16Z"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_201_3969"
                    >
                      <rect
                        fill="white"
                        height="24"
                        width="24"
                      />
                    </clippath>
                  </defs>
                </svg>
              </i>
            </button>
          </span>
        </li>
        <li
          aria-labelledby="react-aria-:r8:"
          class="flex group gap-2 items-center justify-between relative px-2 h-full box-border rounded-small subpixel-antialiased cursor-pointer tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[focus-visible=true]:dark:ring-offset-background-content1 data-[hover=true]:transition-colors data-[hover=true]:bg-default data-[hover=true]:text-default-foreground data-[selectable=true]:focus:bg-default data-[selectable=true]:focus:text-default-foreground py-0 w-full bg-white dark:bg-indigo-light shadow-light-card dark:shadow-none"
          data-key="3"
          id="react-aria-:r0:-option-3"
          role="option"
          tabindex="-1"
        >
          <span
            class="flex-1 text-small font-normal truncate flex justify-between items-center px-4 py-3"
            data-label="true"
            id="react-aria-:r8:"
          >
            <div
              class="inline-flex items-center justify-center gap-2 rounded-small outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              tabindex="-1"
            >
              <span
                class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-tiny bg-default text-default-foreground rounded-full w-[46px] h-[46px]"
                tabindex="-1"
              >
                <img
                  alt="avatar"
                  class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
                  src="/default-avatar.webp"
                />
              </span>
              <div
                class="inline-flex flex-col items-start"
              >
                <span
                  class="text-inherit text-medium font-bold"
                />
                <span
                  class="text-tiny font-medium text-secondary"
                >
                  Product Designer
                </span>
              </div>
            </div>
            <button
              class="disabled:cursor-not-allowed disabled:opacity-80"
              disabled=""
            >
              <i
                class="w-6 h-6 flex justify-center items-center hover:cursor-pointer fill-secondary"
              >
                <svg
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_201_3969)"
                  >
                    <path
                      d="M12 8C13.1 8 14 7.1 14 6C14 4.9 13.1 4 12 4C10.9 4 10 4.9 10 6C10 7.1 10.9 8 12 8ZM12 10C10.9 10 10 10.9 10 12C10 13.1 10.9 14 12 14C13.1 14 14 13.1 14 12C14 10.9 13.1 10 12 10ZM12 16C10.9 16 10 16.9 10 18C10 19.1 10.9 20 12 20C13.1 20 14 19.1 14 18C14 16.9 13.1 16 12 16Z"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_201_3969"
                    >
                      <rect
                        fill="white"
                        height="24"
                        width="24"
                      />
                    </clippath>
                  </defs>
                </svg>
              </i>
            </button>
          </span>
        </li>
        <li
          aria-labelledby="react-aria-:ra:"
          class="flex group gap-2 items-center justify-between relative px-2 h-full box-border rounded-small subpixel-antialiased cursor-pointer tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 data-[focus-visible=true]:dark:ring-offset-background-content1 data-[hover=true]:transition-colors data-[hover=true]:bg-default data-[hover=true]:text-default-foreground data-[selectable=true]:focus:bg-default data-[selectable=true]:focus:text-default-foreground py-0 w-full bg-white dark:bg-indigo-light shadow-light-card dark:shadow-none"
          data-key="4"
          id="react-aria-:r0:-option-4"
          role="option"
          tabindex="-1"
        >
          <span
            class="flex-1 text-small font-normal truncate flex justify-between items-center px-4 py-3"
            data-label="true"
            id="react-aria-:ra:"
          >
            <div
              class="inline-flex items-center justify-center gap-2 rounded-small outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2"
              tabindex="-1"
            >
              <span
                class="flex relative justify-center items-center box-border overflow-hidden align-middle z-0 outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-tiny bg-default text-default-foreground rounded-full w-[46px] h-[46px]"
                tabindex="-1"
              >
                <img
                  alt="avatar"
                  class="flex object-cover w-full h-full transition-opacity !duration-500 opacity-0 data-[loaded=true]:opacity-100"
                  src="/default-avatar.webp"
                />
              </span>
              <div
                class="inline-flex flex-col items-start"
              >
                <span
                  class="text-inherit text-medium font-bold"
                />
                <span
                  class="text-tiny font-medium text-secondary"
                >
                  Junior Graphic Designer
                </span>
              </div>
            </div>
            <button
              class="disabled:cursor-not-allowed disabled:opacity-80"
              disabled=""
            >
              <i
                class="w-6 h-6 flex justify-center items-center hover:cursor-pointer fill-secondary"
              >
                <svg
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    clip-path="url(#clip0_201_3969)"
                  >
                    <path
                      d="M12 8C13.1 8 14 7.1 14 6C14 4.9 13.1 4 12 4C10.9 4 10 4.9 10 6C10 7.1 10.9 8 12 8ZM12 10C10.9 10 10 10.9 10 12C10 13.1 10.9 14 12 14C13.1 14 14 13.1 14 12C14 10.9 13.1 10 12 10ZM12 16C10.9 16 10 16.9 10 18C10 19.1 10.9 20 12 20C13.1 20 14 19.1 14 18C14 16.9 13.1 16 12 16Z"
                    />
                  </g>
                  <defs>
                    <clippath
                      id="clip0_201_3969"
                    >
                      <rect
                        fill="white"
                        height="24"
                        width="24"
                      />
                    </clippath>
                  </defs>
                </svg>
              </i>
            </button>
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
